File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/users/mailFolders/singleValueExtendedProperties/SingleValueExtendedPropertiesResponse.java
package graphjavav4.utilities.users.mailFolders.singleValueExtendedProperties;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import graphjavav4.utilities.users.SingleValueLegacyExtendedProperty;
import java.util.function.BiConsumer;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Objects;
public class SingleValueExtendedPropertiesResponse implements Parsable {
    @javax.annotation.Nullable
    public List<SingleValueLegacyExtendedProperty> value;
    @javax.annotation.Nullable
    public String nextLink;
    /**
     * Serialiazes information the current object
     * @param writer Serialization writer to use to serialize this model
     * @return a void
     */
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeCollectionOfObjectValues("value", value);
        writer.writeStringValue("nextLink", nextLink);
    }
    /**
     * The serialization information for the current model
     * @return a Map<String, BiConsumer<T, ParseNode>>
     */
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getDeserializeFields() {
        final Map<String, BiConsumer<T, ParseNode>> fields = new HashMap<>(2);
        fields.put("value", (o, n) -> { ((SingleValueExtendedPropertiesResponse)o).value = n.getCollectionOfObjectValues(SingleValueLegacyExtendedProperty.class); });
        fields.put("nextLink", (o, n) -> { ((SingleValueExtendedPropertiesResponse)o).nextLink = n.getStringValue(); });
        return fields;
    }
}

File: msgraph-mail/dotnet/Users/MailFolders/SingleValueExtendedProperties/SingleValueExtendedPropertiesResponse.cs
using Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.Linq;
namespace Graphdotnetv4.Users.MailFolders.SingleValueExtendedProperties {
    public class SingleValueExtendedPropertiesResponse : IParsable<SingleValueExtendedPropertiesResponse> {
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; private set; } = new Dictionary<string, object>();
        /// <summary>The serialization information for the current model</summary>
        public IDictionary<string, Action<SingleValueExtendedPropertiesResponse, IParseNode>> DeserializeFields => new Dictionary<string, Action<SingleValueExtendedPropertiesResponse, IParseNode>> {
            {
                "nextLink", (o,n) => { o.NextLink = n.GetStringValue(); }
            },
            {
                "value", (o,n) => { o.Value = n.GetCollectionOfObjectValues<SingleValueLegacyExtendedProperty>().ToList(); }
            },
        };
        public string NextLink { get; set; }
        public List<SingleValueLegacyExtendedProperty> Value { get; set; }
        /// <summary>
        /// Serialiazes information the current object
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        /// </summary>
        public void Serialize(ISerializationWriter writer) {
            writer.WriteStringValue("nextLink", NextLink);
            writer.WriteCollectionOfObjectValues<SingleValueLegacyExtendedProperty>("value", Value);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}
